ntbs-service/Services/AuthorizationService.cs

```diff
@@ -29,7 +35,39 @@ public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Aler
             return userTbServiceCodes.Contains(alert.TbServiceCode);
         }
 
-        public async Task<bool> CanEdit(ClaimsPrincipal user, Notification notification)
+        public IEnumerable<NotificationBannerModel> SetFullAccessOnNotificationBanners(
+            IEnumerable<NotificationBannerModel> notificationBanners,
+            ClaimsPrincipal user)
+        {
+            return notificationBanners.Select(async n =>
+            {
+                if(n.NotificationStatus != NotificationStatus.Legacy)
+                {
+                    var fullAccess = await CanEditBannerModel(user, n);
+                    n.FullAccess = fullAccess;
+                }
+                return n;
+            })
+            .Select(n => n.Result);

```

Comment:
The name of the function would imply we're *modifying* the banners we're passing in, rather than creating new ones - this is not what the method currently does. Perhaps we want something along the lines of:
```
public IEnumerable<NotificationBannerModel> SetFullAccessOnNotificationBanners(
            IEnumerable<NotificationBannerModel> notificationBanners,
            ClaimsPrincipal user)
        {
           var tasks = notificationBanners.ForEach(async n =>
            {
                if (n.NotificationStatus != NotificationStatus.Legacy)
                {
                    n.FullAccess = await CanEditBannerModel(user, n);
                }
            });
            Task.WaitAll(tasks);
        }
```

Appreciate that this is perhaps contrary to my usual suggestions of making things more functional, but in this case it seems quite wasteful to create a whole new set of results, especially when the in and out are the same type!